boards: update arduino_due to use gpio_sam driver, remove gpio_atmel_sam3 driver

mnkp opened the pull request:
Update configuration of arduino_due board to use a common gpio_sam driver rather than sam3x specific gpio_atmel_sam3. This PR also removes, now obsolete, gpio_atmel_sam3 driver.

mnkp committed:
soc: add DTS PIO module nodes for sam3x series

This commit adds DTS PIO nodes for sam3x series SoCs.

Signed-off-by: Piotr Mienkowski <piotr.mienkowski@gmail.com>

mnkp committed:
drivers: fix #ifdef in gpio_sam driver

- prefix SOC_SERIES_ defines with CONFIG_
- exclude pull down configuration on sam3x series

Signed-off-by: Piotr Mienkowski <piotr.mienkowski@gmail.com>

mnkp committed:
boards: update arduino_due to use gpio_sam driver

Update configuration of arduino_due board to use a common gpio_sam
driver rather than sam3x specific gpio_atmel_sam3. The gpio_atmel_sam3
driver is going to be deprecated.

Signed-off-by: Piotr Mienkowski <piotr.mienkowski@gmail.com>

mnkp committed:
boards: add "User LED" to arduino_due board DTS

This commit adds user yellow LED to arduino_due board DTS.

Signed-off-by: Piotr Mienkowski <piotr.mienkowski@gmail.com>

galak commented:
I feel like this is a case in which we could just remove the driver.

mnkp commented:
Yes, I think we could simply remove the driver. This PR lays the groundwork. Should I add a commit that removes gpio_atmel_sam3 or it's better to do it in a separate PR some time later?

galak commented:
> Yes, I think we could simply remove the driver. This PR lays the groundwork. Should I add a commit that removes gpio_atmel_sam3 or it's better to do it in a separate PR some time later?

I'm fine if you want to remove it now as part of this PR.

mnkp committed:
drivers: remove gpio_atmel_sam3 driver

The gpio_atmel_sam3 is no longer required since the gpio_sam, common
SAM family GPIO driver, provides improved functionality.

Signed-off-by: Piotr Mienkowski <piotr.mienkowski@gmail.com>
